File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Angular/src/app/interfaces/response/updatemyselfresponse.ts
import { User } from "../user";
import { State } from "../state";

export interface UpdateMyselfResponse {
    user: User | null;
    errorOnlyUpdatesToYourselfAreAllowed: boolean;
    success: boolean;
    errorAuthentication: boolean;
    errorItemNotFound: boolean;
    state: State | null;
}

File: BeltmanSoftwareDesign.Angular/BeltmanSoftwareDesign.Shared/ResponseJsons/UpdateMyselfResponse.cs
using BeltmanSoftwareDesign.Shared.Jsons;

namespace BeltmanSoftwareDesign.Shared.ResponseJsons
{
    public class UpdateMyselfResponse : Response
    {
        public User? User {  get; set; }
        public bool ErrorOnlyUpdatesToYourselfAreAllowed { get; set; }
    }
}
